pennylane/templates/embeddings/amplitude.py

```diff
@@ -216,17 +141,82 @@ def circuit(f=None):
+
+        wires = Wires(wires)
+        self.pad_with = pad_with
+        self.normalize = normalize
+
+        features = self._preprocess(features, wires, pad_with, normalize)
+        super().__init__(features, wires=wires, do_queue=do_queue)
+
+    def expand(self):
+
+        with qml.tape.QuantumTape() as tape:
+            QubitStateVector(self.parameters[0], wires=self.wires)
+
+        return tape
 
-    # pad is replaced with the more verbose pad_with
-    if pad is not None:
-        warnings.warn(
-            "The pad argument will be replaced by the pad_with option in future versions of PennyLane.",
-            PendingDeprecationWarning,
-        )
-        if pad_with is None:
-            pad_with = pad
+    @staticmethod
+    def _preprocess(features, wires, pad_with, normalize):
+        """Validate and pre-process inputs as follows:
 
-    features = _preprocess(features, wires, pad_with, normalize)
+        * Check that the features tensor is one-dimensional.
+        * If pad_with is None, check that the first dimension of the features tensor
+          has length :math:`2^n` where :math:`n` is the number of qubits. Else check that the
+          first dimension of the features tensor is not larger than :math:`2^n` and pad features with value if necessary.
+        * If normalize is false, check that first dimension of features is normalised to one. Else, normalise the
+          features tensor.
+        """
+
+        shape = qml.math.shape(features)
+
+        # check shape
+        if len(shape) != 1:
+            raise ValueError(f"Features must be a one-dimensional tensor; got shape {shape}.")
+
+        n_features = shape[0]
+        if pad_with is None and n_features != 2 ** len(wires):
+            raise ValueError(
+                f"Features must be of length {2 ** len(wires)}; got length {n_features}. "
+                f"Use the 'pad' argument for automated padding."
+            )
+
+        if pad_with is not None and n_features > 2 ** len(wires):
+            raise ValueError(
+                f"Features must be of length {2 ** len(wires)} or "
+                f"smaller to be padded; got length {n_features}."
+            )
 
-    QubitStateVector(features, wires=wires)
+        # pad
+        if pad_with is not None and n_features < 2 ** len(wires):
+            padding = [pad_with] * (2 ** len(wires) - n_features)
+            features = qml.math.concatenate([features, padding], axis=0)
+
+        # normalize
+        norm = qml.math.sum(qml.math.abs(features) ** 2)
+
+        if not qml.math.allclose(norm, 1.0, atol=TOLERANCE):
+            if normalize or pad_with:
+                features = features / np.sqrt(norm)
+            else:
+                raise ValueError(
+                    f"Features must be a vector of length 1.0; got length {norm}."
+                    "Use 'normalize=True' to automatically normalize."
+                )
+
+        features = qml.math.cast(features, np.complex128)
```

Comment:
Added this line -  it turns out that tf cannot deal with real inputs, but this is a typical use case for the embedding.